src/staticanalysis/bot/static_analysis_bot/workflow.py

```diff
@@ -279,28 +279,33 @@ def detect_issues(self, analyzers, revision, is_before=False):
         for analyzer_class in analyzers:
             # Build analyzer
             analyzer = analyzer_class()
-
-            if is_before:
-                if analyzer.can_run_before_patch():
-                    # Run analyzer on revision and store generated issues
-                    logger.info('Run {} with no patch applied'.format(analyzer_class.__name__))
-                    analyzer_issues = analyzer.run(revision)
+            analyzer_issues = []
+            try:
+                if is_before:
+                    if analyzer.can_run_before_patch():
+                        # Run analyzer on revision and store generated issues
+                        logger.info('Run {} with no patch applied'.format(analyzer_class.__name__))
+                        analyzer_issues = analyzer.run(revision)
+                    else:
+                        logger.info('Skipped running {} with no patch applied'.format(analyzer_class.__name__))
+                        continue
                 else:
-                    logger.info('Skipped running {} with no patch applied'.format(analyzer_class.__name__))
-                    continue
-            else:
-                logger.info('Run {}'.format(analyzer_class.__name__))
-                analyzer_issues = analyzer.run(revision)
-
-            # Clean up any uncommitted changes left behind by this analyzer.
-            self.hg.revert(settings.repo_dir.encode('utf-8'), all=True)
-            logger.info('Reverted all uncommitted changes in repo', rev=self.hg.identify())
-
-            # Compute line hashes now before anything else changes the code.
-            for issue in analyzer_issues:
-                issue.build_lines_hash()
-
-            issues += analyzer_issues
+                    logger.info('Run {}'.format(analyzer_class.__name__))
+                    analyzer_issues = analyzer.run(revision)
+            except AnalysisException as ex:
+                # Log the error to Sentry
+                logger.error('Analyzer {} thrown exceptions during runtime.'.format(analyzer_class.__name__))
+                logger.error(ex.message)
+            finally:
+                # Clean up any uncommitted changes left behind by this analyzer.
+                self.hg.revert(settings.repo_dir.encode('utf-8'), all=True)
+                logger.info('Reverted all uncommitted changes in repo', rev=self.hg.identify())
+
+                # Compute line hashes now before anything else changes the code.
```

Comment:
I'd move this part in the try block, so you don't need to define `analyzer_issues` before as an empty list

Reply:
Sure good point.

Reply:
Oh wait, maybe we have to compute the line hashes *after* the revert? @La0 do you remember?

Reply:
[Some work was done](https://github.com/mozilla/release-services/pull/1745) last december around this.

I'm not sure computing after the revert changes anything

Reply:
Actually, in that thread of discussion, @jankeromnes fixed an issue by making the cleanup be done before the hash calculation: https://github.com/mozilla/release-services/pull/1745#issuecomment-447839046.

So maybe we have to keep the same order here.